components/p3a/brave_p3a_service.cc

```diff
@@ -306,6 +325,18 @@ void BraveP3AService::OnHistogramChanged(base::StringPiece histogram_name,
       base::StatisticsRecorder::FindHistogram(histogram_name)->SnapshotDelta();
   DCHECK(!samples->Iterator()->Done());
 
+  // Shortcut for the special values, see |kSuspendedMetricValue|
+  // description for details.
+  if (IsSuspendedMetric(histogram_name, sample)) {
+    base::PostTask(FROM_HERE, {content::BrowserThread::UI},
+                   base::BindOnce(&BraveP3AService::OnHistogramChangedOnUI,
+                                  this,
+                                  histogram_name,
+                                  kSuspendedMetricValue,
+                                  kSuspendedMetricBucket));
+    return;
+  }
+
```

Comment:
I think it would be good to locate this code at the start of this method.
So, we don't need to find samples if suspended.

Reply:
nope, we still have to "scrub" sample from histogram storage. `SnapshotDelta()` actually extracts it. Otherwise we wouldn't be able to it get new samples